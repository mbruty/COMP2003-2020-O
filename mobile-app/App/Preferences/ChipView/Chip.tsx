import React, { Dispatch, SetStateAction } from "react";
import { Text, View, StyleSheet, TouchableOpacity, Image } from "react-native";

interface Props {
  title: string;
  index: number;
  enabled: boolean;
  setTouched: (touched: boolean, index: number) => void;
}

export const Chip: React.FC<Props> = (props) => {
  const viewStyles: Array<any> = [styles.container];

  viewStyles.push(props.enabled ? styles.active : styles.inactive);

  const textStyle = props.enabled ? styles.textActive : styles.textDisabled;

  const icon = props.enabled
    ? require("../../../resources/icons/heart.png")
    : require("../../../resources/icons/baseline_not_interested_black_48.png");
  return (
    <TouchableOpacity
      style={{ height: 50 }}
      onPress={() => props.setTouched(!props.enabled, props.index)}
    >
      <View style={viewStyles}>
        <Image style={styles.icon} source={icon}></Image>
        <Text style={textStyle}>{props.title}</Text>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    margin: 5,
    padding: 8,
    paddingRight: 15,
    borderRadius: 20,
    flex: 1,
    flexDirection: "row",
    elevation:10,
  },
  active: {
    backgroundColor: "#FD4040",
  },
  inactive: {
    backgroundColor: "#FFF",
  },
  textActive: {
    marginTop: 2,
    paddingLeft: 5,
    color: "#FFF",
  },
  textDisabled: {
    marginTop: 2,
    paddingLeft: 5,
    color: "#707070",
  },
  icon: {
    width: 24,
    height: 24,
    marginLeft: 5,
  },
});
