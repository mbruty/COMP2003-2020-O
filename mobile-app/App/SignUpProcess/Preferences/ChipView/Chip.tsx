import React from "react";
import { Text, View, StyleSheet, TouchableOpacity, Image } from "react-native";
import { CONSTANT_STYLES } from "../../shared/constants";

interface Props {
  title: string;
  index: number;
  enabled: boolean;
  setTouched: (touched: boolean, index: number) => void;
}

export const Chip: React.FC<Props> = (props) => {
  const viewStyles: Array<any> = [styles.container];

  // Add disabled / enabled colour to the view styles
  viewStyles.push(
    props.enabled ? CONSTANT_STYLES.BG_RED : CONSTANT_STYLES.BG_WHITE
  );

  const textStyle = props.enabled ? [styles.textActive, CONSTANT_STYLES.TXT_WHITE] : [styles.textDisabled, CONSTANT_STYLES.TXT_DEFAULT];

  // Set the icon to the enabled / disabled one
  const icon = props.enabled
    ? require("./Chip/baseline_check_circle_outline_white_18.png")
    : require("../shared/baseline_not_interested_black_48.png");
  return (
    <TouchableOpacity
      style={{ height: 50 }}
      onPress={() => props.setTouched(!props.enabled, props.index)}
    >
      <View style={viewStyles}>
        <Image style={styles.icon} source={icon}></Image>
        <Text allowFontScaling={false} style={textStyle}>
          {props.title}
        </Text>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    margin: 5,
    padding: 8,
    paddingRight: 15,
    borderRadius: 20,
    flex: 1,
    flexDirection: "row",
    elevation: 10,
  },
  textActive: {
    marginTop: 2,
    paddingLeft: 5,
  },
  textDisabled: {
    marginTop: 2,
    paddingLeft: 5,
  },
  icon: {
    width: 24,
    height: 24,
    marginLeft: 5,
  },
});
