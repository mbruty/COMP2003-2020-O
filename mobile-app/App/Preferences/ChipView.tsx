import React, { useState } from "react";
import { Text, View, StyleSheet } from "react-native";
import { Chip } from "./ChipView/Chip";
interface Props {
  chipNameList: Array<string>;
}
export const ChipView: React.FC<Props> = (props) => {
  const [enabledArray, setEnabledArray] = useState<Array<boolean>>(
    new Array<boolean>(props.chipNameList.length).fill(true)
  );
  const setTouched = (touched: boolean, index: number) => {
    let enabled = [...enabledArray];
    enabled[index] = touched;
    setEnabledArray(enabled);
  };
  return (
    <View style={styles.container}>
      {props.chipNameList.map((chip, index) => (
        <Chip
          title={chip}
          enabled={enabledArray[index]}
          index={index}
          setTouched={setTouched}
        />
      ))}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "row",
    justifyContent: "flex-start",
    marginLeft: 20,
    flexWrap: "wrap"
  }
})
