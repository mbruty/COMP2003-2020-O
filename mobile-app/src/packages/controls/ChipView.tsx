import React from "react";
import { View, StyleSheet } from "react-native";
import { Chip } from "./Chip";
interface Props {
  chipNameList: Array<string>;
  setTouched: (index: number) => void;
}
const ChipView: React.FC<Props> = (props) => {
  return (
    <View style={styles.container}>
      {props.chipNameList.map((chip, index) => (
        <Chip
          key={`chip-${index}`}
          title={chip}
          index={index}
          setTouched={props.setTouched}
        />
      ))}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    display: "flex",
    flexDirection: "row",
    justifyContent: "flex-start",
    marginLeft: 20,
    flexWrap: "wrap",
    alignSelf: "flex-start",
    marginBottom: 30,
  },
});

export default ChipView;
