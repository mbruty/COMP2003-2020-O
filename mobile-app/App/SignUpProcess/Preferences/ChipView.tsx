import React, { useState } from "react";
import { Text, View, StyleSheet } from "react-native";
import { Chip } from "./ChipView/Chip";
interface Props {
  chipNameList: Array<string>;
  enabledArray: Array<boolean>;
  setTouched: (touched: boolean, index: number) => void;
}
export const ChipView: React.FC<Props> = (props) => {
  

  return (
    <View style={styles.container}>
      {props.chipNameList.map((chip, index) => (
        <Chip
          key={`chip-${index}`}
          title={chip}
          enabled={props.enabledArray[index]}
          index={index}
          setTouched={props.setTouched}
        />
      ))}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "row",
    justifyContent: "flex-start",
    marginLeft: 20,
    flexWrap: "wrap",
  },
});
