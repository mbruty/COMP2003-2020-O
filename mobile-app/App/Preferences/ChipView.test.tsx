import React from "react";
import renderer from "react-test-renderer";
import { ChipView } from "./ChipView";

test("Renders Chip List", () => {
  const chipList = renderer
    .create(<ChipView chipNameList={["Chip One", "Chip Two"]} />)
    .toJSON();
  expect(chipList).toMatchSnapshot();
});
