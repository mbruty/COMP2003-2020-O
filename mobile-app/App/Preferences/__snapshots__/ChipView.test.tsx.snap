// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Chip List 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "flexWrap": "wrap",
      "justifyContent": "flex-start",
      "marginLeft": 20,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": 50,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 20,
            "elevation": 10,
            "flex": 1,
            "flexDirection": "row",
            "margin": 5,
            "padding": 8,
            "paddingRight": 15,
          },
          Object {
            "backgroundColor": "#FD4040",
          },
        ]
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 24,
            "marginLeft": 5,
            "width": 24,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#FFF",
            "marginTop": 2,
            "paddingLeft": 5,
          }
        }
      >
        Chip One
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": 50,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 20,
            "elevation": 10,
            "flex": 1,
            "flexDirection": "row",
            "margin": 5,
            "padding": 8,
            "paddingRight": 15,
          },
          Object {
            "backgroundColor": "#FD4040",
          },
        ]
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 24,
            "marginLeft": 5,
            "width": 24,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#FFF",
            "marginTop": 2,
            "paddingLeft": 5,
          }
        }
      >
        Chip Two
      </Text>
    </View>
  </View>
</View>
`;
